aten/src/ATen/native/TensorShape.cpp

```diff
@@ -767,6 +767,9 @@ Tensor make_qtensor(const Tensor& self, IntArrayRef size, IntArrayRef stride, Qu
 }
 
 Tensor as_strided_tensorimpl(const Tensor& self, IntArrayRef size, IntArrayRef stride, optional<int64_t> storage_offset_) {
+  if (self.is_zerotensor()) {
+    return at::_efficientzerotensor(self.sizes(), self.options());
```

Comment:
Should this return a Tensor of the size `size` and not self's ?

Do we have tests for these?